build.gradle

```diff
@@ -136,7 +135,13 @@ subprojects {
                 ext.mbeddrBuildCounter = GitBasedVersioning.getGitCommitCount()
             }
 
-            ext.mbeddrBuildNumber = GitBasedVersioning.getVersion(mbeddrBuild, mbeddrMajor, mbeddrMinor, mbeddrBuildCounter as int)
+            if(mbeddrBuild == "master" || mbeddrBuild.matches(/(maintenance|mps)[\/-](mps)?\d+(\.\d+)*/)) {
+                ext.mbeddrBuildNumber = GitBasedVersioning.getVersion(mbeddrBuild, mbeddrMajor, mbeddrMinor, mbeddrBuildCounter as int)
+            } else {
+                // use same logic as in all other platforms for snapshot publications
+                ext.mbeddrBuildNumber = GitBasedVersioning.getVersionWithCount(mbeddrMajor, mbeddrMinor, mbeddrBuildCounter as int) + "-SNAPSHOT"
```

Comment:
Shouldn't we add branch name to the snapshot version as well?

If not, then we don't need to use `getVersion` in the 'then' branch either, can use `getVersionWithCount` in both cases (and refactor the if).

Reply:
You are right, I think a branch name should be part of the snapshot:

I'll add to this to keep that information.  
`GitBasedVersioning.getVersion(mbeddrBuild, mbeddrMajor, mbeddrMinor, mbeddrBuildCounter as int) + "-SNAPSHOT"`

However the snapshot publication is still not working. :(

Reply:
We need to add publishing to mbeddr / Pull Requests on TeamCity. But in order to do that we need to make sure it doesn't publish snapshots on GitHub.

Reply:
Regarding the branch name agein: I Just checked _GitBasedVersioning.getVersionWithCount()_, and it should actaully determine the branch name [itself](https://github.com/mbeddr/mps-gradle-plugin/blob/master/src/main/groovy/de/itemis/mps/gradle/GitBasedVersioning.groovy#L77). However, it looks like it is not working properly for our TC config.